tests/remote_resources/xtcp_ping_pong.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/python
+import socket
+import random
+import string
+import time
+import argparse
+import sys
+from multiprocessing import Pool
+
+def print_errors(failure_list):
+    for process_failures in failure_list:
+        if len(process_failures) > 0:
+            print ''.join(process_failures)
+
+    print "Lost_packets: {}".format(sum(map(len, failure_list)))
+
+def format_message(message, width):
+    return '\n  '.join([message[i:i+width] for i in range(0, len(message), width)])
+
+# Limit reconnect attempts
+def attempt_connect(failures, process_port, sock=None):
+    for _ in range(3):
+        try:
+            sock = socket.socket(socket.AF_INET, args.protocol)
+            sock.settimeout(5) # seconds
+            sock.connect((args.ip, process_port))
+            return (failures, sock)
+
+        except socket.error as err:
+            failures.append('ERROR: Could not connect to remote port\n' +
+                            '  Local port:    {}\n'.format(sock.getsockname()[1]) +
+                            '  Remote port:   {}\n'.format(process_port) +
+                            '  Error message: {}\n'.format(err)
+                            )
+
+    return (failures, None)
+
+def process_test(process_port):
+    random.seed()
+    tests_to_perform = args.packets
+    failures = []
+
+    # Prevents bombarding the board
```

Comment:
Is this delay to avoid simultaneous connection attempts? Which part of the system struggles to cope with this?